### PR TITLE
Improve onTokenTransfer hooks documentation

### PR DESCRIPTION
**Pull request by jurgenwerk**
Ticket: CS-3350

This PR aims improve the documentation on how we use the ERC677 `onTokenTransfer` callback in our contracts. It explains the params and offers a short description about its intent and params, and points to the spot in the readme where one can get more details on the handlers. 

~~TODO: checking with @tintinthong to see how to explain the ignored reward related empty param and if we can remove it~~

### CODE REVIEW

File: contracts/action-handlers/AddRewardRuleHandler.sol [hunk 1]
```diff
@@ -41,9 +41,26 @@ contract AddRewardRuleHandler is Ownable, Versionable {
...
+   * @dev onTokenTransfer(ERC677) - this is the ERC677 token transfer callback.
+   *
+   * This adds the reward rule to the reward manager.
+   *
+   * See AddRewardRuleHandler in README for more information.
+   *
+   * @param from the token sender (should be the action dispatcher)
+   * @param data encoded as: (
+   *  address prepaidCard,
+   *  uint256 ???,
```
Comment (jurgenwerk) [inline contracts/action-handlers/AddRewardRuleHandler.sol:54]: @tintinthong here in [this line](https://github.com/cardstack/card-pay-protocol/blob/main/test/RewardManager-test.js#L486) it says it doesn't matter what this is - how come we can't just remove it?

**Review by alex-cardstack — APPROVED** [review]
Excellent work, thank you @jurgenwerk

File: contracts/action-handlers/AddRewardRuleHandler.sol (hunk 1 above)
Comment (tintinthong) [reply contracts/action-handlers/AddRewardRuleHandler.sol:54]: Owh this was in the test right. Yea just remove all instance of "it doesn't matter what it is". I think it was me simulating a transferAndCall but the handler doesn't care about the amount.

**Review by tintinthong — APPROVED** [review]
LGTM

**Comment by jurgenwerk** [comment]
@alex-cardstack @tintinthong added a commit with the explanation of the ignored param. I think this is now good to merge!

**Review by alex-cardstack — APPROVED** [review]

**Pull request merged**